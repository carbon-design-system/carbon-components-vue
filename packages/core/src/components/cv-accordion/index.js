import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
