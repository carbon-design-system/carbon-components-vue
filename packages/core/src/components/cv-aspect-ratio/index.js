import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
