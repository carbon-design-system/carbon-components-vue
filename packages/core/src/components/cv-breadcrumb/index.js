import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
