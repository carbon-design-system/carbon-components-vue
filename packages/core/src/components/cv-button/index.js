import CvButton from './cv-button';
import CvIconButton from './cv-icon-button';
import CvButtonSkeleton from './cv-button-skeleton';
import CvButtonSet from './cv-button-set';

export { CvButton, CvIconButton, CvButtonSkeleton, CvButtonSet };
