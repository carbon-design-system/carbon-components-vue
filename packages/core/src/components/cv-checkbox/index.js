import CvCheckbox from './cv-checkbox';
import CvCheckboxSkeleton from './cv-checkbox-skeleton';
export { CvCheckbox, CvCheckboxSkeleton };
