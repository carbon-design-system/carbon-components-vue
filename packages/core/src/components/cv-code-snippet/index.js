import CvCodeSnippet from './cv-code-snippet';
import CvCodeSnippetSkeleton from './cv-code-snippet-skeleton';
export { CvCodeSnippet, CvCodeSnippetSkeleton };
