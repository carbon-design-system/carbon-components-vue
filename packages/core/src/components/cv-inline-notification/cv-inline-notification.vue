<template>
  <div
    data-notification
    :class="[
      'cv-inline-notification',
      `${carbonPrefix}--inline-notification`,
      `${carbonPrefix}--inline-notification--${kind.toLowerCase()}`,
      {
        [`${carbonPrefix}--inline-notification--low-contrast`]: lowContrast,
      },
    ]"
    v-on="$listeners"
    :role="isAlert ? 'alert' : false"
    :aria-live="!isAlert ? 'polite' : false"
  >
    <div :class="`${carbonPrefix}--inline-notification__details`">
      <component :is="icon" :class="`${carbonPrefix}--inline-notification__icon`" />
      <div :class="`${carbonPrefix}--inline-notification__text-wrapper`">
        <p :class="`${carbonPrefix}--inline-notification__title`">{{ title }}</p>
        <p :class="`${carbonPrefix}--inline-notification__subtitle`" v-html="subTitle"></p>
      </div>
    </div>
    <button
      v-if="actionLabel"
      @click="$emit('action')"
      :class="[
        `${carbonPrefix}--inline-notification__action-button`,
        `${carbonPrefix}--btn`,
        `${carbonPrefix}--btn--sm`,
        `${carbonPrefix}--btn--ghost`,
      ]"
      type="button"
    >
      {{ actionLabel }}
    </button>
    <button
      type="button"
      :aria-label="closeAriaLabel"
      data-notification-btn
      :class="`${carbonPrefix}--inline-notification__close-button`"
      @click="$emit('close')"
    >
      <Close20 :class="`${carbonPrefix}--inline-notification__close-icon`" />
    </button>
  </div>
</template>

<script>
import { notificationMixin, carbonPrefixMixin } from '../../mixins';
import ErrorFilled20 from '@carbon/icons-vue/es/error--filled/20';
import CheckmarkFilled20 from '@carbon/icons-vue/es/checkmark--filled/20';
import WarningFilled20 from '@carbon/icons-vue/es/warning--filled/20';
import Close20 from '@carbon/icons-vue/es/close/20';
import InformationFilled20 from '@carbon/icons-vue/es/information--filled/20';

export default {
  name: 'CvInlineNotification',
  components: { Close20 },
  mixins: [notificationMixin, carbonPrefixMixin],
  props: {
    actionLabel: { type: String, default: '' },
    closeAriaLabel: { type: String, default: 'Dismiss notification' },
    kind: {
      type: String,
      default: 'info',
      validator: val => ['error', 'info', 'warning', 'success'].includes(val),
    },
    lowContrast: Boolean,
  },
  computed: {
    icon() {
      switch (this.kind) {
        case 'error':
          return ErrorFilled20;
        case 'warning':
          return WarningFilled20;
        case 'success':
          return CheckmarkFilled20;
        case 'info':
          return InformationFilled20;
        default:
          return '';
      }
    },
  },
};
</script>
