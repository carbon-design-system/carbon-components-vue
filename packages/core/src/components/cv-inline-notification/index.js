import CvInlineNotification from './cv-inline-notification';

export { CvInlineNotification };
