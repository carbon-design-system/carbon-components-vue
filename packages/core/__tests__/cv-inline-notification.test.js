import { shallowMount as shallow } from '@vue/test-utils';
import { testComponent } from './_helpers';
import { CvInlineNotification } from '@/components/cv-inline-notification';
import ErrorFilled20 from '@carbon/icons-vue/es/error--filled/20';
import CheckmarkFilled20 from '@carbon/icons-vue/es/checkmark--filled/20';
import WarningFilled20 from '@carbon/icons-vue/es/warning--filled/20';

describe('CvInlineNotification', () => {
  const kinds = ['error', 'info', 'warning', 'success'];

  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(CvInlineNotification, ['lowContrast'], Boolean);
  testComponent.propsAreType(CvInlineNotification, ['actionLabel', 'closeAriaLabel', 'kind'], String);
  testComponent.propsHaveDefault(CvInlineNotification, ['closeAriaLabel', 'kind', 'actionLabel']);

  it('`kind` prop validator works as expected', () => {
    const propsData = { lowContrast: false };

    const wrapper = shallow(CvInlineNotification, { propsData });
    for (const kind of kinds) {
      expect(wrapper.vm.$options.props.kind.validator && wrapper.vm.$options.props.kind.validator(kind)).toBeTruthy();
    }

    // suppress the error message from the kind validator
    const consoleError = console.error;
    console.error = jest.fn();

    expect(wrapper.vm.$options.props.kind.validator && wrapper.vm.$options.props.kind.validator('TEST')).toBeFalsy();

    // restore
    console.error = consoleError;
  });

  // ***************
  // SNAPSHOT TESTS
  // ***************
  it('should render correctly', () => {
    const propsData = { lowContrast: false, kind: '', actionLabel: 'test action label' };
    for (const kind of kinds) {
      propsData.kind = kind;
      const wrapper = shallow(CvInlineNotification, { propsData });
      expect(wrapper.html()).toMatchSnapshot();
    }
  });

  it('should render correctly when low contrast is used', () => {
    const propsData = { lowContrast: true, kind: '', actionLabel: 'test action label' };
    for (const kind of kinds) {
      propsData.kind = kind;
      const wrapper = shallow(CvInlineNotification, { propsData });
      expect(wrapper.html()).toMatchSnapshot();
    }
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************

  it('`icon` is computed correctly', () => {
    const kindToIconMapping = {
      error: ErrorFilled20,
      warning: WarningFilled20,
      success: CheckmarkFilled20,
      info: '',
    };
    const propsData = { actionLabel: 'test action label', lowContrast: false, kind: '' };
    for (const kind of kinds) {
      propsData.kind = kind;
      const wrapper = shallow(CvInlineNotification, { propsData });
      expect(wrapper.vm.icon).toEqual(kindToIconMapping[kind]);
    }
  });

  it('should emit close event when close button is clicked', () => {
    const propsData = { actionLabel: 'test action label', lowContrast: false };
    const wrapper = shallow(CvInlineNotification, { propsData });
    wrapper.find('.bx--inline-notification__close-button').trigger('click');
    expect(wrapper.emitted().close).toBeTruthy();
  });

  it('should emit action event when action button is clicked', () => {
    const propsData = { actionLabel: 'test action label', lowContrast: false };
    const wrapper = shallow(CvInlineNotification, { propsData });
    wrapper.find('.bx--inline-notification__action-button').trigger('click');
    expect(wrapper.emitted().action).toBeTruthy();
  });
});
