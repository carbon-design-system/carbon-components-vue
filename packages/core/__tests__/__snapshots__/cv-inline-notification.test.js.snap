// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvInlineNotification should not render close button when hideCloseButton is true 1`] = `
<div data-notification="" role="alert" class="cv-inline-notification bx--inline-notification bx--inline-notification--error">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button>
  <!---->
</div>
`;

exports[`CvInlineNotification should not render close button when hideCloseButton is true 2`] = `
<div data-notification="" aria-live="polite" class="cv-inline-notification bx--inline-notification bx--inline-notification--info">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button>
  <!---->
</div>
`;

exports[`CvInlineNotification should not render close button when hideCloseButton is true 3`] = `
<div data-notification="" role="alert" class="cv-inline-notification bx--inline-notification bx--inline-notification--warning">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button>
  <!---->
</div>
`;

exports[`CvInlineNotification should not render close button when hideCloseButton is true 4`] = `
<div data-notification="" aria-live="polite" class="cv-inline-notification bx--inline-notification bx--inline-notification--success">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button>
  <!---->
</div>
`;

exports[`CvInlineNotification should render correctly 1`] = `
<div data-notification="" role="alert" class="cv-inline-notification bx--inline-notification bx--inline-notification--error">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button> <button type="button" aria-label="Dismiss notification" data-notification-btn="" class="bx--inline-notification__close-button">
    <close20-stub class="bx--inline-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvInlineNotification should render correctly 2`] = `
<div data-notification="" aria-live="polite" class="cv-inline-notification bx--inline-notification bx--inline-notification--info">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button> <button type="button" aria-label="Dismiss notification" data-notification-btn="" class="bx--inline-notification__close-button">
    <close20-stub class="bx--inline-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvInlineNotification should render correctly 3`] = `
<div data-notification="" role="alert" class="cv-inline-notification bx--inline-notification bx--inline-notification--warning">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button> <button type="button" aria-label="Dismiss notification" data-notification-btn="" class="bx--inline-notification__close-button">
    <close20-stub class="bx--inline-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvInlineNotification should render correctly 4`] = `
<div data-notification="" aria-live="polite" class="cv-inline-notification bx--inline-notification bx--inline-notification--success">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button> <button type="button" aria-label="Dismiss notification" data-notification-btn="" class="bx--inline-notification__close-button">
    <close20-stub class="bx--inline-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvInlineNotification should render correctly when low contrast is used 1`] = `
<div data-notification="" role="alert" class="cv-inline-notification bx--inline-notification bx--inline-notification--error bx--inline-notification--low-contrast">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button> <button type="button" aria-label="Dismiss notification" data-notification-btn="" class="bx--inline-notification__close-button">
    <close20-stub class="bx--inline-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvInlineNotification should render correctly when low contrast is used 2`] = `
<div data-notification="" aria-live="polite" class="cv-inline-notification bx--inline-notification bx--inline-notification--info bx--inline-notification--low-contrast">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button> <button type="button" aria-label="Dismiss notification" data-notification-btn="" class="bx--inline-notification__close-button">
    <close20-stub class="bx--inline-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvInlineNotification should render correctly when low contrast is used 3`] = `
<div data-notification="" role="alert" class="cv-inline-notification bx--inline-notification bx--inline-notification--warning bx--inline-notification--low-contrast">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button> <button type="button" aria-label="Dismiss notification" data-notification-btn="" class="bx--inline-notification__close-button">
    <close20-stub class="bx--inline-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvInlineNotification should render correctly when low contrast is used 4`] = `
<div data-notification="" aria-live="polite" class="cv-inline-notification bx--inline-notification bx--inline-notification--success bx--inline-notification--low-contrast">
  <div class="bx--inline-notification__details">
    <anonymous-stub class="bx--inline-notification__icon"></anonymous-stub>
    <div class="bx--inline-notification__text-wrapper">
      <p class="bx--inline-notification__title"></p>
      <p class="bx--inline-notification__subtitle"></p>
    </div>
  </div> <button type="button" class="bx--inline-notification__action-button bx--btn bx--btn--sm bx--btn--ghost">
    test action label
  </button> <button type="button" aria-label="Dismiss notification" data-notification-btn="" class="bx--inline-notification__close-button">
    <close20-stub class="bx--inline-notification__close-icon"></close20-stub>
  </button>
</div>
`;
